source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -559,7 +559,16 @@ void Filter::onReceiveMessage(std::unique_ptr<ProcessingResponse>&& r) {
 
   // Check whether the server is asking to extend the timer.
   if (response->has_override_message_timeout()) {
-    onNewTimeout(DurationUtil::durationToMilliseconds(response->override_message_timeout()));
+    // The override_message_timeout in response may be too big for duration, which leads
+    // to exception been thrown during durationToMilliseconds() check.
+    // Needs to properly handle this case.
+    try {
+      onNewTimeout(DurationUtil::durationToMilliseconds(response->override_message_timeout()));
+    } catch (const DurationUtil::OutOfRangeException& e) {
```

Comment:
throwing C++ exceptions is not allowed on the dataplane due to concerns about netspecre as well as their error prone nature that can lead to abnormal process terminations. For example if the `durationToMilliseconds` is change din the future to throw another type of exception (i.e. some future DurationUtil::MalformedDurationException) but this catch statement is not adjusted accordingly, this can lead to crash as the new exception would not be caught. Java compiler for example guards against this types of changes, but C++ does not. Which is why we prefer using return values to indicate error conditions.

For this PR you can factor out the functions for parsing duration into the one that throw errors and return errors, and use the functions that return errors here.

Reply:
Thanks for the comments!

I added a new utility function as below to be called in data plane and refactored the underline validate function:
static uint64_t durationToMillisecondsNoThrow(const ProtobufWkt::Duration& duration, bool& error);

Reply:
I am trying to change the Duration override_message_timeout field into a uint32 override_timeout_ms field to avoid calling such durationToMilliseconds() function in dataplane.

Reply:
Actually, changing the Duration override_message_timeout field into a uint32 override_timeout_ms field is causing more confusing in APIs. So, let's stay with what we already have.  Also there are already a couple of such NoThrow version of utility functions existing in the same file, so adding a new one is reasonable.

Reply:
Done!